### PR TITLE
Add progressive downsampling for nearest neighbor frames.

### PR DESCRIPTION
**Pull request by Philbrick**

### CODE REVIEW

File: src/nearestneighborframe.h [hunk 1]
```diff
@@ -31,15 +32,24 @@ class NearestNeighborFrame : public Frame {
...
-  // frameWidht, frameHeight_ - size frame needs to be scaled to
+  // frameWidth, frameHeight_ - size frame needs to be scaled to
   // compression - type of compression
+  // quality - compression quality setting
+  // store_raw_bytes - store raw version of frame pixels in frame in addition
+  //                   to compressed pixel bytes. Required for progressive
+  //                   downsampling.
+  // frame_region_reader - frame reader for raw frame data from prior level.
+  //                       Used to generate downsamples directly from prior
+  //                       downsampled leve.
```
Comment (danielbeaudreau) [inline src/nearestneighborframe.h]: typo

File: src/wsiToDcm.cpp [hunk 2]
```diff
@@ -367,7 +376,12 @@ int WsiToDcm::dicomizeTiff(
       adapative_stop_downsampling = true;
     }
     pool.join();
-    generated_dicom_files.clear();
+    if (preferProgressiveDownsampling) {
+      higher_magnifcation_dicom_files.set_dicom_files(
+                                             std::move(generated_dicom_files));
```
Comment (danielbeaudreau) [inline src/wsiToDcm.cpp:381]: Does generated_dicom_files also need to be cleared here to prepare for the next iteration or does the std::move take care of that?

File: src/nearestneighborframe.h (hunk 1 above)
Comment (Philbrick) [reply src/nearestneighborframe.h]: fixed

File: src/wsiToDcm.cpp (hunk 2 above)
Comment (Philbrick) [reply src/wsiToDcm.cpp:381]: The move takes care of it.  Move does a efficient move of the memory.  The effect is generated_dicom_files have no files in them after the move and they have been moved to the into higher magnification.

random search, example: 
https://stackoverflow.com/questions/12613428/stl-vector-moving-all-elements-of-a-vector

Comment (danielbeaudreau) [reply src/wsiToDcm.cpp:381]: Thanks.

**Pull request merged**